com.hazelcast.jet.impl.connector.JmsSourceIntegrationTest_RabbitMQ.stressTest_exactlyOnce_forceful_durableTopic

_master_ (commit 07d7d7a4b3c5cecada1a5e9a8b35d571360c399e)

Failed on Nightly build: http://jenkins.hazelcast.com/job/jet-oss-master-nightly/143/testReport/junit/com.hazelcast.jet.impl.connector/JmsSourceIntegrationTest_RabbitMQ/stressTest_exactlyOnce_forceful_durableTopic/

It fails regularly even locally.

Stacktrace:
```
java.lang.AssertionError: jobId=0428-b17c-ae53-0001 expected:<RUNNING> but was:<FAILED>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at com.hazelcast.jet.core.JetTestSupport.lambda$assertJobStatusEventually$1(JetTestSupport.java:192)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:191)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:148)
	at com.hazelcast.jet.impl.connector.JmsSourceIntegrationTestBase.stressTest(JmsSourceIntegrationTestBase.java:335)
	at com.hazelcast.jet.impl.connector.JmsSourceIntegrationTestBase.stressTest_exactlyOnce_forceful_durableTopic(JmsSourceIntegrationTestBase.java:294)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
2020-04-06 23:19:26,567 [ERROR] [hz.hardcore_albattani.cached.thread-21] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Execution of job '0428-b17c-ae53-0001', execution 0428-b17c-ae54-0001 failed
	Start time: 2020-04-06T23:19:26.541
	Duration: 26 ms
	For further details enable JobConfig.storeMetricsAfterJobCompletion
com.hazelcast.jet.JetException: Exception in ProcessorTasklet{jmsTopicSource(?)#0}: java.lang.AbstractMethodError: com.rabbitmq.jms.client.RMQSession.createSharedDurableConsumer(Ljavax/jms/Topic;Ljava/lang/String;)Ljavax/jms/MessageConsumer;
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:304) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_241]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_241]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_241]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_241]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_241]
Caused by: java.lang.AbstractMethodError: com.rabbitmq.jms.client.RMQSession.createSharedDurableConsumer(Ljavax/jms/Topic;Ljava/lang/String;)Ljavax/jms/MessageConsumer;
	at com.hazelcast.jet.impl.connector.JmsSourceIntegrationTestBase.lambda$stressTest$69d1b45c$1(JmsSourceIntegrationTestBase.java:307) ~[test-classes/:?]
	at com.hazelcast.function.FunctionEx.apply(FunctionEx.java:47) ~[hazelcast-4.0.1-SNAPSHOT.jar:4.0.1-SNAPSHOT]
	at com.hazelcast.jet.impl.connector.StreamJmsP.init(StreamJmsP.java:115) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.init(AbstractProcessor.java:83) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.init(ProcessorWrapper.java:80) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.init(ProcessorTasklet.java:228) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:289) ~[classes/:?]
	... 5 more
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/JmsSourceIntegrationTest_RabbitMQ.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/JmsSourceIntegrationTest_RabbitMQ.java
@@ -41,4 +41,9 @@ public class JmsSourceIntegrationTest_RabbitMQ extends JmsSourceIntegrationTestB
     protected SupplierEx<ConnectionFactory> getConnectionFactory() {
         return FACTORY_SUPPLIER;
     }
+
+    @Override
+    public void stressTest_exactlyOnce_forceful_durableTopic() {
+        // ignore this test for RabbitMQ, it doesn't support JMS 2.0
+    }
 }

